internal/services/machinelearning/machine_learning_datastore_blobstorage_resource.go

```diff
@@ -0,0 +1,268 @@
+
+			"workspace_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.WorkspaceID,
+			},
+
+			"storage_account_name": {
+				Type:             pluginsdk.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc:     validation.StringIsNotEmpty,
+			},
+
+			"container_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"description": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"is_default": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"service_data_auth_identity": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(datastore.ServiceDataAccessAuthIdentityNone),
+					string(datastore.ServiceDataAccessAuthIdentityWorkspaceSystemAssignedIdentity),
+					string(datastore.ServiceDataAccessAuthIdentityWorkspaceUserAssignedIdentity),
+				},
+					false),
+				Default: string(datastore.ServiceDataAccessAuthIdentityNone),
+			},
+
+			"account_key": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				ExactlyOneOf: []string{"account_key", "shared_access_signature"},
+			},
+
+			"shared_access_signature": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				AtLeastOneOf: []string{"account_key", "shared_access_signature"},
+			},
+
+			"tags": commonschema.TagsForceNew(),
+		},
+	}
+	return resource
+}
+
+func resourceMachineLearningDataStoreCreateOrUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MachineLearning.DatastoreClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	workspaceId, err := workspaces.ParseWorkspaceID(d.Get("workspace_id").(string))
+	if err != nil {
+		return err
+	}
+
+	id := datastore.NewDataStoreID(subscriptionId, workspaceId.ResourceGroupName, workspaceId.WorkspaceName, d.Get("name").(string))
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, id)
+		if err != nil {
+			if !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+		}
+		if !response.WasNotFound(existing.HttpResponse) {
+			return tf.ImportAsExistsError("azurerm_machine_learning_datastore_blobstorage", id.ID())
+		}
+	}
+
+	datastoreRaw := datastore.DatastoreResource{
+		Name: utils.String(d.Get("name").(string)),
+		Type: utils.ToPtr(string(datastore.DatastoreTypeAzureBlob)),
+	}
+
+	prop := expandBlobStorage(d)
```

Comment:
The schema is flat, so I don't quite see what we're expanding here or gaining by creating the props payload in a separate function?